Integrar análisis con GenAI para reportes de prueba

dianazepeda3 opened the pull request:
Se introduce una nueva funcionalidad en el módulo Publisher para generar análisis automáticos de los resultados de las suites de prueba utilizando un LLM local a través de Ollama

dianazepeda3 committed:
Integrar análisis con GenAI para reportes de prueba

Se introduce una nueva funcionalidad en el módulo Publisher para generar análisis automáticos de los resultados de las suites de prueba utilizando un LLM local a través de Ollama

S41nz commented on tbtaf/publisher/report_ai.py:
```diff
@@ -0,0 +1,34 @@
+import json
+
+# Constantes para la configuración de Ollama
+OLLAMA_API_URL = "http://localhost:11434/api/generate"
```
Lets load this from an env variable, then if not defined, then we should error out.

S41nz reviewed (changes requested):
As mentioned in some comments, please migrate both implementation and documentation to english for consistency.

Thanks

dianazepeda3 committed:
Redesign GenAI module using design patterns

Refactored the GenAI integration for improved modularity and extensibility based on architectural feedback.
- Decoupled AI logic into a dedicated 'ai' package.
- Implemented the Factory and Interface design patterns.
- Moved API configuration to environment variables.

S41nz commented on tbtaf/publisher/reporte_de_bugs.md:
```diff
@@ -0,0 +1,157 @@
+
+This document serves as a formal record of the technical issues (bugs) identified and resolved during the setup and development of the TBTAF framework enhancement project. Each bug is documented in a format similar to a GitHub Issue to facilitate traceability.
+
+---
```
Migrate this to GitHub issues and remove the file from the commit

S41nz commented on tbtaf/publisher/generators/PDFReportGenerator.py:
```diff
@@ -6,15 +6,47 @@
+    execution reports in PDF format.
     '''
+    # Define the base path as a class attribute
+    TEST_CODE_BASE_PATH = "../test/smoke"
```
Same comment as HTMLReportGenerator abotu the test code path

dianazepeda3 committed:
Refactor Report Generators and Improve Processes

- Created a `RAGBasedReportGenerator` parent class for shared AI logic.
- Configured `TEST_CODE_BASE_PATH` to use an environment variable with a default fallback.
- Removed the in-code bug reporter; bugs are now tracked as formal issues.
